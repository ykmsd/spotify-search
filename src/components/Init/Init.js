import React from 'react';

const Init = () => {
  return (
    <div>
      <p>You need to login to your Spotify account to use this app.</p>
    </div>
  );
};

export default Init;
