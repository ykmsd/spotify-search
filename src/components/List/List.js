// @flow
import React from 'react';
import { Row, Col, Card } from 'antd';
import { Link } from 'react-router-dom';

const { Meta } = Card;

type Props = {
  items: Array<Object>,
}

const List = ({ items } : Props) => {
  return (
    <div>
      <Row gutter={16}>
        {
          items && 
            items.map(item => (
              <Link to={`/${item.albums.id}`} key={item.albums.id}>
                <Col span={4} style={{ marginBottom: '20px' }}>
                  <Card
                    cover={<img src={item.albums.images[0].url} alt={`${item.albums.name}`} />}
                    hoverable
                  >
                    <Meta
                      title={item.albums.name}
                      description={item.albums.artists[0].name}
                    />
                  </Card>
                </Col>
              </Link>
            ))
        }
      </Row>
    </div>
  );
};

export default List;
