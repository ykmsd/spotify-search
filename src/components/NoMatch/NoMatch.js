import React from 'react';

const NoMatch = () => {
  return (
    <div>
      <p>Something went wrong. Please try it again later.</p>
    </div>
  );
};

export default NoMatch;
