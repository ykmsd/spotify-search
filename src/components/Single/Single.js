import * as React from 'react';
import { Row, Col, Button, Icon } from 'antd';

type Props = {
  routeProps: Object,
  items: Array<Object>,
}

const Single = ({ routeProps, items } : Props) => {
  const { albumId } = routeProps.match.params;
  const i = items.findIndex(album => album.albums.id === albumId);
  const album = items[i];
  const { albums } = album;
  return (
    <div>
      <Row gutter={16} type="flex" justify="center" style={{ marginBottom: '40px' }}>
        <Col span={6}>
          <iframe src={`https://open.spotify.com/embed?uri=${albums.uri}&theme=white`} width="300" height="380" frameBorder="0" allowtransparency="true" title="Spotify Player" />
        </Col>
        <Col span={6}>
          <h2>{albums.name}</h2>
          <h3>{albums.artists[0].name}</h3>
          <img src={albums.images[1].url} alt={albums.name} />             
        </Col>
      </Row>
      <Button type="primary" onClick={routeProps.history.goBack}>
        <Icon type="left" />Go back
      </Button>
    </div>
  );
};

export default Single;
