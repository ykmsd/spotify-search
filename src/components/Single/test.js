// /* global expect, it, describe, beforeEach, jest */

// import React from 'react';
// import { shallow, mount } from 'enzyme';
// import Single from './Single';

// describe('Single', () => {
//   let component;

//   const props = {
//     items: [
//       {
//         albums: {
//           name: 'reputation',
//           artists: ['Taylor Swift'],
//           id: '4fW1sFeE43nuZlAw2xtmC3',
//           images: [{
//             url: 'https://i.scdn.co/image/39cb05d373ddb356dd8f5357810ab35fdec9f6e1',
//           }],
//         },
//       },
//     ],
//     routeProps: {
//       match: {
//         params: {
//           albumId: "2x9mn9i1j3xJzz7GEH8We2"
//         },
//       },
//     },
//   };
//   beforeEach(() => {
//     component = mount(<Single {...props} />);
//   });

//   it('Should render successfully', () => {
//     // expect(component.exists()).toEqual(true);
//   });

//   it('Should show one image', () => {
//     // expect(component.find('img').length).to.have.lengthOf(1);
//   });
// });
