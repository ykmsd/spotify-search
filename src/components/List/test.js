/* global expect, it, describe, beforeEach, jest */

import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

describe('List', () => {
  let component;

  const items = [
    {
      albums: {
        name: 'reputation',
        artists: ['Taylor Swift'],
        id: '4fW1sFeE43nuZlAw2xtmC3',
        images: [{
          url: 'https://i.scdn.co/image/39cb05d373ddb356dd8f5357810ab35fdec9f6e1',
        }],
      },
    },
  ];

  beforeEach(() => {
    component = shallow(<List items={items} />);
  });


  it('Should render successfully', () => {
    expect(component.exists()).toEqual(true);
  });

  // it('Should render the same number of images as the items length', () => {
  //   expect(component.find('img').length).toEqual(items.length);
  // });
});
