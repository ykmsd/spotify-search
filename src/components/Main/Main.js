// @flow

import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Single from '../Single/Single';
import List from '../List/List';
import Init from '../Init/Init';
import NoMatch from '../NoMatch/NoMatch';

type Props = {
  searchAlbum: Function,
  setToken: Function,
  token: string,
  albums: Array<Object>,
  hasErrored: boolean,
  isLoading: boolean,
}

const Main = (props: Props) => {
  return (
    <div>
      <Switch>
        <Route path="/" component={Init} exact />
        <Route path="/callback" render={routeProps => (<List routeProps={routeProps} {...props} />)} />
        <Route path="/view/:albumId" render={routeProps => (<Single routeProps={routeProps} albums={props.albums} />)} />
        <Route component={NoMatch} />
      </Switch>
    </div>
  );
};

export default Main;
