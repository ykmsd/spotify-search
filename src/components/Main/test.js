/* global describe, expect, it, jest */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Main from './Main';
import Single from '../Single/Single';
import List from '../List/List';
import Init from '../Init/Init';
import NoMatch from '../NoMatch/NoMatch';

describe('Main', () => {

  const props = {
    searchAlbum: jest.fn(),
    setToken: jest.fn(),
    token: 'abc',
    albums: [{ name: 'Taylor' }, { name: 'Gorillaz' }],
    hasErrored: false,
    isLoading: false,
  };

  const SingleProps = {
    routeProps: {
      history: {
        goBack: jest.fn(),
      },
      match: {
        params: '12',
      },
    },
    albums: [{ name: 'Taylor' }, { name: 'Gorillaz' }],
  };

  const ListProps = {
    albums: [{ name: 'Taylor' }, { name: 'Gorillaz' }],
    hasErrored: false,
    isLoading: false,
    searchAlbum: jest.fn(),
    token: 'abc',
  };
  

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter><Main {...props} /></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('callback routing', () => {
    const tree = renderer.create(<MemoryRouter initialEntries={['/callback']}><List {...ListProps} /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('view albumId routing', () => {
    const tree = renderer.create(<MemoryRouter initialEntries={['/view/:albumId']}><Single {...SingleProps} /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('no match routing', () => {
    const tree = renderer.create(<MemoryRouter initialEntries={['/404']}><NoMatch /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
