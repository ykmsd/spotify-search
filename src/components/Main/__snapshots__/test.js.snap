// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main callback routing 1`] = `
<div>
  <div>
    
    <form
      className="ant-form ant-form-inline"
      onSubmit={[Function]}
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    >
      <div
        className="ant-row ant-form-item"
        style={undefined}
      >
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control "
          >
            <input
              className="ant-input search-input"
              disabled={false}
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Search by Artist name"
              style={
                Object {
                  "width": "200px",
                }
              }
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={undefined}
      >
        <div
          className="ant-form-item-control-wrapper"
        >
          <div
            className="ant-form-item-control "
          >
            <button
              className="ant-btn search-button ant-btn-primary"
              onClick={[Function]}
              type="submit"
            >
              <span>
                Search
              </span>
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
  <div
    className="ant-row"
    style={undefined}
  >
    <div
      className="ant-col-xs-12 ant-col-sm-10 ant-col-md-8 ant-col-lg-6 ant-col-xl-4"
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    >
      <a
        href="/view/undefined"
        onClick={[Function]}
      >
        <div
          className="ant-card ant-card-bordered ant-card-hoverable"
        >
          <div
            className="ant-card-cover"
          >
            <img
              alt="undefined"
              src={undefined}
              style={
                Object {
                  "maxHeight": "280px",
                }
              }
            />
          </div>
          <div
            className="ant-card-body"
            style={undefined}
          >
            <div>
              <div
                className="ant-card-meta"
              />
            </div>
          </div>
        </div>
      </a>
    </div>
    <div
      className="ant-col-xs-12 ant-col-sm-10 ant-col-md-8 ant-col-lg-6 ant-col-xl-4"
      style={
        Object {
          "marginBottom": "20px",
        }
      }
    >
      <a
        href="/view/undefined"
        onClick={[Function]}
      >
        <div
          className="ant-card ant-card-bordered ant-card-hoverable"
        >
          <div
            className="ant-card-cover"
          >
            <img
              alt="undefined"
              src={undefined}
              style={
                Object {
                  "maxHeight": "280px",
                }
              }
            />
          </div>
          <div
            className="ant-card-body"
            style={undefined}
          >
            <div>
              <div
                className="ant-card-meta"
              />
            </div>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`Main no match routing 1`] = `
<div>
  <p>
    Something went wrong. Please try it again later.
  </p>
</div>
`;

exports[`Main renders correctly 1`] = `
<div>
  <div>
    <p>
      You need to login to your Spotify account to use this app.
    </p>
  </div>
</div>
`;

exports[`Main view albumId routing 1`] = `
<div>
  <div
    className="ant-row-flex ant-row-flex-center"
    style={
      Object {
        "marginBottom": "40px",
        "marginLeft": -8,
        "marginRight": -8,
      }
    }
  >
    <div
      className="ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-6 ant-col-xl-6"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <iframe
        allowtransparency="true"
        frameBorder="0"
        height="380"
        src="https://open.spotify.com/embed?uri=undefined&theme=white"
        title="Spotify Player"
        width="300"
      />
    </div>
    <div
      className="ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-6 ant-col-xl-6"
      style={
        Object {
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
    >
      <h2 />
      <h3 />
      <img
        alt={undefined}
        src={undefined}
      />
    </div>
  </div>
  <button
    className="ant-btn ant-btn-primary"
    onClick={[Function]}
    type="button"
  >
    <i
      className="anticon anticon-left"
    />
    <span>
      Go back
    </span>
  </button>
</div>
`;
